### PR TITLE
#15145: Add support for multi-device tensors in grouped convolution weight preprocessing

### PR DESCRIPTION
**Pull request by esmalTT**
### Ticket
https://github.com/tenstorrent/tt-metal/issues/15145

### Summary

Currently - convolution weight preprocessing for grouped convolutions fails when using multi-device tensors. This PR adds support for using multi-device tensors with grouped convolutions. 

Specifically, the issue is that `conv_group_weight_zero_pad_helper` does not handle the case where the input is multi-device or `OwnedStorage`. I have modified `conv_group_weight_zero_pad_helper` to handle these types of inputs.

I added `ttnn.conv2d` tests that cover multi-device convolutions for `groups >= 1` which covers the case where inputs are muti-device and `OwnedStorage`. Ialso fixed formatting in `tensor_utils.cpp`.

### Checklist
- [x] Post commit CI passes: [link](https://github.com/tenstorrent/tt-metal/actions/runs/11787725443)
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Comment by esmalTT** [comment]
Added `ttnn.conv2d` tests that cover multi-device convolutions for `groups >= 1`

**Review by cfjchu — APPROVED** [review]

**Review by ayerofieiev-tt — COMMENTED** [review]
Does the new test also convers tensor with `OwnedStorage`?

File: tests/ttnn/unit_tests/operations/test_new_conv2d.py [hunk 1]
```diff
@@ -288,7 +303,7 @@ def run_conv_with_split(
                 torch_bias_zeroes_tensor, weights_dtype if weights_dtype != ttnn.bfloat8_b else ttnn.float32
             )
         torch_input_tensor = torch.permute(split_input_tensors[i], (0, 2, 3, 1))
-        tt_input_tensor = ttnn.from_torch(torch_input_tensor, ttnn.bfloat16)
+        tt_input_tensor = ttnn.from_torch(torch_input_tensor, ttnn.bfloat16, device=device)
```
Comment (pavlejosipovic) [inline tests/ttnn/unit_tests/operations/test_new_conv2d.py:306]: why do we push tensor to device here?
In run_conv we we don't push the tensor to device before the conv2d starts?

Comment (esmalTT) [reply tests/ttnn/unit_tests/operations/test_new_conv2d.py:306]: This was a mistake, this `run_conv_with_split` should not be modified. Will fix.

**Comment by esmalTT** [comment]
> Does the new test also convers tensor with `OwnedStorage`?

@ayerofieiev-tt Yes. Multi-device tensors are `OwnedStorage`.

**Review by pavlejosipovic — APPROVED** [review]

**Review by mywoodstock — APPROVED** [review]

**Pull request merged**